### PR TITLE
test(e2e): Add test to connect to target via its name and scope

### PR DESCRIPTION
**Pull request by stasryzhov**
This PR adds a test to verify connection to targets via a combination of the `-target-name` and `-target-scope-name` options.

The PR also:
- adds an optional parameter to the `boundary.CreateNewProjectCli` to be able to optionally set custom project name
- adds support of the `target.WithName` option for the `boundary.CreateNewTargetCli`

### CODE REVIEW

File: testing/internal/e2e/tests/base/target_tcp_connect_test.go [hunk 1]
```diff
@@ -65,6 +65,59 @@ func TestCliTcpTargetConnectTargetBasic(t *testing.T) {
...
+	boundary.AddHostToHostSetCli(t, ctx, newHostSetId, newHostId)
+	newTargetId := boundary.CreateNewTargetCli(t, ctx, newProjectId, c.TargetPort, target.WithName(testTargetName))
+	boundary.AddHostSourceToTargetCli(t, ctx, newTargetId, newHostSetId)
+
+	// Connect to target via target and scope names, and print host's IP address
+	output := e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"connect",
+			"-target-name", testTargetName,
+			"-target-scope-name", testProjectName,
```
Comment (stasryzhov) [inline testing/internal/e2e/tests/base/target_tcp_connect_test.go:101]: Connecting to target via its name and scope

Comment (moduli) [reply testing/internal/e2e/tests/base/target_tcp_connect_test.go:101]: How do these options work? Do you always have to provide both options?

File: testing/internal/e2e/boundary/scope.go [hunk 2]
```diff
@@ -83,3 +93,29 @@ func CreateNewProjectCli(t testing.TB, ctx context.Context, orgId string) string
 	t.Logf("Created Project Id: %s", newProjectId)
 	return newProjectId
 }
+
+// getScopeOpts iterates the inbound ScopeOptions and returns a struct
+func getScopeOpts(opt ...ScopeOption) scopeOptions {
```
Comment (moduli) [inline testing/internal/e2e/boundary/scope.go:98]: For targets, we use boundary's target options already defined in the application code. Is there not a corresponding one for scopes?

File: testing/internal/e2e/tests/base/target_tcp_connect_test.go (hunk 1 above)
Comment (stasryzhov) [reply testing/internal/e2e/tests/base/target_tcp_connect_test.go:101]: Yes. The `-target-name` should be used in conjunction with either `-target-scope-name` or `-target-scope-id`

Documentation: [Selecting Targets](https://developer.hashicorp.com/boundary/docs/hcp/get-started/connect-to-target#selecting-targets)

So we may want to add a target connect attempt with `-target-name` and `-target-scope-id` as well

Comment (moduli) [reply testing/internal/e2e/tests/base/target_tcp_connect_test.go:101]: Sounds good. Should we also modify the names to include some other special characters?

**Review by johanbrandhorst — APPROVED** [review]
Approved pending Michael's comments

File: testing/internal/e2e/boundary/scope.go (hunk 2 above)
Comment (stasryzhov) [reply testing/internal/e2e/boundary/scope.go:98]: I haven't found it in the code base. If I missed that, let me know, please 🙏

Comment (moduli) [reply testing/internal/e2e/boundary/scope.go:98]: I'm not super familiar with this. I did find this:https://github.com/hashicorp/boundary-enterprise/blob/main/api/scopes/option.gen.go?

Comment (moduli) [reply testing/internal/e2e/boundary/scope.go:98]: Hm. I might be confusing them with the API options (https://github.com/hashicorp/boundary-enterprise/blob/main/api/scopes/option.gen.go).

**Review by moduli — APPROVED** [review]

**Pull request merged**